xWorkflows/Xworkflow1/go/kubecost-event/main.go

```diff
@@ -0,0 +1,160 @@
+func GetUnclaimedVolumes() ([]PersistentVolume, error) {
+	// Make a GET request to the Kubecost API
+	response, err := http.Get(kubecostAPIURL)
+	if err != nil {
+		return nil, fmt.Errorf("error making GET request to Kubecost API: %w", err)
+	}
+	defer response.Body.Close()
+
+	// Read response data
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %w", err)
+	}
+
+	// Parse JSON response
+	var data map[string]interface{}
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshalling JSON response: %w", err)
+	}
+
+	// Ensure that 'items' key exists and is of the correct type
+	items, ok := data["items"].([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("error: 'items' field is not a slice of interfaces")
+	}
+
+	// Extract volume names and statuses
+	var unclaimedPVs []PersistentVolume
+	for _, item := range items {
+		itemMap, ok := item.(map[string]interface{})
+		if !ok {
+			continue // Skip if item is not a map
+		}
+
+		// Ensure 'status' is a map
+		status, ok := itemMap["status"].(map[string]interface{})
+		if !ok {
+			continue // Skip items without a "status" field or if it's not a map
+		}
+
+		// Check if the phase is "Available"
+		if phase, ok := status["phase"].(string); ok && phase == "Available" {
+			// Ensure 'name' is a string
+			if metadata, ok := itemMap["metadata"].(map[string]interface{}); ok {
+				if name, ok := metadata["name"].(string); ok {
+					pv := PersistentVolume{
+						Name:   name,
+						Status: "Available",
+					}
+					unclaimedPVs = append(unclaimedPVs, pv)
+				}
+			}
+		}
+	}
+
+	return unclaimedPVs, nil
+}
+
+func SaveVolumesToMongoDB(collection *mongo.Collection, volumes []PersistentVolume) error {
+	// Create a unique index on the 'name' field
+	indexOptions := options.Index().SetUnique(true)
+	index := mongo.IndexModel{
+		Keys:    map[string]interface{}{"name": 1}, // 1 for ascending, -1 for descending
+		Options: indexOptions,
+	}
+
+	// Create the index on the collection
+	_, err := collection.Indexes().CreateOne(context.Background(), index)
+	if err != nil {
+		return fmt.Errorf("error creating index: %w", err)
+	}
+
+	// Insert volumes into MongoDB
+	for _, volume := range volumes {
+		_, err := collection.InsertOne(context.Background(), volume)
+		if err != nil {
+			if mongo.IsDuplicateKeyError(err) {
+				fmt.Printf("Document with name '%s' already exists.\n", volume.Name)
+				// Handle duplicate entry gracefully
+				continue
+			}
+			return fmt.Errorf("error inserting document into MongoDB: %w", err)
+		}
+	}
+	return nil
+}
+
+func main() {
+	// MongoDB connection URI
+	mongoURI := "mongodb://xworkflow-mongodb-0.xworkflow-mongodb.default.svc.cluster.local:27017,xworkflow-mongodb-1.xworkflow-mongodb.default.svc.cluster.local:27017,xworkflow-mongodb-2.xworkflow-mongodb.default.svc.cluster.local:27017/?replicaSet=xworkflowReplSet"
```

Comment:
Suggestion: using env var here and passing value using pod.spec container env variables

Reply:
added env vars for these values